src/Elastic.Apm/Model/Span.cs

```diff
@@ -91,7 +94,19 @@ private Span(double duration, string id, string name, string parentId)
 					enclosingTransaction.SpanCount.IncrementStarted();
 
 					if (captureStackTraceOnStart)
-						_stackFrames = new EnhancedStackTrace(new StackTrace(true)).GetFrames();
+					{
+						var stackTrace = new StackTrace(true);
+						try
+						{
+							// I saw EnhancedStackTrace throwing exceptions in some environments
+							// therefore we try-catch and fall back to a non-demystified call stack.
```

Comment:
worth opening an issue for?

Reply:
As discussed over zoom - there is not much to do about it. One known platform is mono and once it happened on Full Framework, which I was unable to repro.

What happens here is that we try to demystify async stack traces with Ben.Demystifier, if it fails with an exception (which would be hard to predict upfront) we just take the existing async callstack as fallback.

One option would be of course to contribute to Ben.Demystifier.

Reply:
If we can get a repro, it'd be worth fixing and contributing. Until then, the current behaviour sounds fine